src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -191,40 +180,26 @@ protected static Task<ImmutableArray<SyntaxToken>> FindMatchingIdentifierTokensA
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
-                if (tokensMatch(state, token, value, cancellationToken))
+                var (matched, reason) = await SymbolsMatchAsync(
+                    symbol, state, token, cancellationToken).ConfigureAwait(false);
+                if (matched)
                 {
-                    var (matched, reason) = await SymbolsMatchAsync(
-                        symbol, state, token, cancellationToken).ConfigureAwait(false);
-                    if (matched)
-                    {
-                        RoslynDebug.Assert(token.Parent != null);
+                    RoslynDebug.Assert(token.Parent != null);
 
-                        var alias = state.Cache.GetAliasInfo(semanticFacts, token, cancellationToken);
+                    var alias = state.Cache.GetAliasInfo(semanticFacts, token, cancellationToken);
 
-                        var location = token.GetLocation();
-                        var symbolUsageInfo = GetSymbolUsageInfo(token.Parent, state, cancellationToken);
+                    var location = token.GetLocation();
+                    var symbolUsageInfo = GetSymbolUsageInfo(token.Parent, state, cancellationToken);
 
-                        locations.Add(new FinderLocation(token.Parent, new ReferenceLocation(
-                            state.Document, alias, location, isImplicit: false,
-                            symbolUsageInfo, GetAdditionalFindUsagesProperties(token.Parent, state), reason)));
-                    }
+                    locations.Add(new FinderLocation(token.Parent, new ReferenceLocation(
+                        state.Document, alias, location, isImplicit: false,
+                        symbolUsageInfo, GetAdditionalFindUsagesProperties(token.Parent, state), reason)));
                 }
             }
 
             return locations.ToImmutable();
         }
 
-        protected static ValueTask<ImmutableArray<FinderLocation>> FindReferencesInTokensAsync(
-            ISymbol symbol,
-            FindReferencesDocumentState state,
-            IEnumerable<SyntaxToken> tokens,
-            Func<FindReferencesDocumentState, SyntaxToken, CancellationToken, bool> tokensMatch,
-            CancellationToken cancellationToken)
-        {
-            return FindReferencesInTokensAsync(
-                symbol, state, tokens, static (state, token, tokensMatch, cancellationToken) => tokensMatch(state, token, cancellationToken), tokensMatch, cancellationToken);
-        }
```

Comment:
lots of varying back and forth helpers, with lambdas capturing delegates, and so forth.